src/ui/atoms/label/index.tsx

```diff
@@ -8,19 +9,24 @@ import { Variant } from 'lib/types';
  */
 
 interface LabelProps extends React.LabelHTMLAttributes<HTMLLabelElement> {
-  text: React.ReactNode;
-  children?: never;
+  children: React.ReactNode;
 }
+const LabelBase: React.FC<LabelProps & Variant> = ({
+  children,
+  variant = 'default',
+}) => <label data-variant={variant}>{children}</label>;
```

Comment:
```suggestion
}) => <label className={className} data-variant={variant}>{children}</label>;
```
иначе стили не применятся.

а вообще такое лучше делать через [`.attrs`](https://styled-components.com/docs/faqs#when-to-use-attrs)